src/processors/src/data-processor.ts

```diff
@@ -388,21 +392,56 @@ export function processKeplerglDataset(
   return Array.isArray(rawData) ? results : results[0];
 }
 
-export const DATASET_HANDLERS: {
-  row: typeof processRowObject;
-  geojson: typeof processGeojson;
-  csv: typeof processCsvData;
-  keplergl: typeof processKeplerglDataset;
-} = {
+/**
+ * Parse a arrow table with geometry columns and return a dataset
+ *
+ * @param arrowTable the arrow table to parse
+ * @returns dataset containing `fields` and `rows` or null
+ */
+export function processArrowTable(arrowBatches: arrow.RecordBatch[]): ProcessorResult | null {
+  if (arrowBatches.length === 0) {
+    return null;
+  }
+  const arrowTable = new arrow.Table(arrowBatches);
+  const fields: Field[] = [];
+
+  // parse fields
+  arrowTable.schema.fields.forEach((field: arrow.Field, index: number) => {
+    const isGeometryColumn = field.metadata.get('ARROW:extension:name')?.startsWith('geoarrow');
```

Comment:
Nit: This means we have knowledge about the GeoArrow extensions both here and in loader.gl/utils? Maybe hard to avoid duplication, but it is always good to try to centralize knowledge about some aspect in one part of the code. I.e. if we could use one of the utils from loaders...

Reply:
Yes, agree. Will expose this in loaders.gl/utils and call it from here.